operator/src/lib.rs

```diff
@@ -131,6 +135,187 @@ fn get_pods_for_node(hashed: &HashMap<String, Vec<Pod>>) -> Vec<Pod> {
+                info!(
+                    "Initial installation, now moving towards version [{}]",
+                    spec_version
+                );
+                self.status = self
+                    .set_upgrading_condition(
+                        &status.conditions,
+                        &format!("Initial installation to version [{:?}]", spec_version),
+                        "InitialInstallation",
+                        ConditionStatus::True,
+                    )
+                    .await?
+                    .status;
+                self.status = self.set_target_version(Some(&spec_version)).await?.status;
+            }
+            (None, Some(target_version)) => {
+                // No current_version but a target_version means we're still doing the initial
+                // installation. Will continue working towards that goal even if another version
+                // was set in the meantime.
+                info!(
+                    "Initial installation, still moving towards version [{}]",
+                    target_version
+                );
+                if &spec_version != target_version {
+                    info!("A new target version ([{}]) was requested while we still do the initial installation to [{}], finishing running upgrade first", spec_version, target_version)
+                }
+                // We do this here to update the observedGeneration if needed
+                self.status = self
+                    .set_upgrading_condition(
+                        &status.conditions,
+                        &format!("Initial installation to version [{:?}]", spec_version),
+                        "InitialInstallation",
+                        ConditionStatus::True,
+                    )
+                    .await?
+                    .status;
+            }
+            (Some(current_version), None) => {
+                // We are at a stable version but have no target_version set. This will be the normal state.
+                // We'll check if there is a different version in spec and if it is will set it in target_version.
+                // TODO: check valid up/downgrades
+                let message;
+                let reason;
+                if current_version.is_upgrade(&spec_version).unwrap() {
+                    message = format!(
+                        "Upgrading from [{:?}] to [{:?}]",
+                        current_version, &spec_version
+                    );
+                    reason = "Upgrading";
+                    self.status = self.set_target_version(Some(&spec_version)).await?.status;
+                } else if current_version.is_downgrade(&spec_version).unwrap() {
+                    message = format!(
+                        "Downgrading from [{:?}] to [{:?}]",
+                        current_version, &spec_version
+                    );
+                    reason = "Downgrading";
```

Comment:
Setting the `Upgrading` condition to `Downgrading` seems a bit strange.... but yeah let's just keep it like that for now. Other than introducing a new condition I don't have another idea.